examples/demo.rs

```diff
@@ -84,6 +89,19 @@ fn main() {
                 });
                 thread_id += 1;
             }
+            "history" => {
+                for (i, entry) in reader.history().enumerate() {
+                    println!("{}: {}", i, entry);
+                }
+            }
+            "save-history" => {
+                if let Err(e) = reader.save_history(HISTORY_FILE) {
+                    eprintln!("Could not save history file {}: {}", HISTORY_FILE, e);
+                }
```

Comment:
`} else {` should all be on the same line. I know, it's kind of a peculiar style, but it's typical for Rust.

Reply:
Fixed. No problem, i'll get it right one of these days. Of course, if studied your code more closely i wouldn't have picked on the style.